pkg/apis/maistra/conversion/extension_providers.go

```diff
@@ -18,34 +19,153 @@ func populateExtensionProvidersValues(in *v2.ControlPlaneSpec, values map[string
+			}
+			if ext.EnvoyExtAuthzHTTP.IncludeAdditionalHeadersInCheck != nil {
+				envoyExtAuthzHTTPValues["includeAdditionalHeadersInCheck"] = mapOfStringToInterface(ext.EnvoyExtAuthzHTTP.IncludeAdditionalHeadersInCheck)
+			}
+			if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck != nil {
+				includeRequestBodyInCheckValues := map[string]interface{}{}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.MaxRequestBytes != nil {
+					includeRequestBodyInCheckValues["maxRequestBytes"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.MaxRequestBytes
+				}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.AllowPartialMessage != nil {
+					includeRequestBodyInCheckValues["allowPartialMessage"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.AllowPartialMessage
+				}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.PackAsBytes != nil {
+					includeRequestBodyInCheckValues["packAsBytes"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.PackAsBytes
+				}
+				envoyExtAuthzHTTPValues["includeRequestBodyInCheck"] = includeRequestBodyInCheckValues
+			}
+			extensionProvidersValues = append(extensionProvidersValues, map[string]interface{}{
+				"name":              ext.Name,
+				"envoyExtAuthzHttp": envoyExtAuthzHTTPValues,
+			})
 		}
 	}
+	if err := setHelmMapSliceValue(values, "meshConfig.extensionProviders", extensionProvidersValues); err != nil {
+		return err
+	}
 	return nil
 }
 
 func populateExtensionProvidersConfig(in *v1.HelmValues, out *v2.ControlPlaneSpec) error {
-	if rawExtProviders, ok, err := in.GetAndRemoveSlice("meshConfig.extensionProviders"); ok {
+	if rawExtProviders, ok, err := in.GetSlice("meshConfig.extensionProviders"); ok {
 		if len(rawExtProviders) == 0 {
 			out.ExtensionProviders = []*v2.ExtensionProviderConfig{}
 		}
 		for _, rawExtProvider := range rawExtProviders {
 			if extProvider, ok := rawExtProvider.(map[string]interface{}); ok {
-				out.ExtensionProviders = append(out.ExtensionProviders, &v2.ExtensionProviderConfig{
-					Name:       extProvider["name"].(string),
-					Prometheus: &v2.ExtensionProviderPrometheusConfig{},
-				})
+				if _, ok := extProvider["prometheus"]; ok {
+					out.ExtensionProviders = append(out.ExtensionProviders, &v2.ExtensionProviderConfig{
+						Name:       extProvider["name"].(string),
+						Prometheus: &v2.ExtensionProviderPrometheusConfig{},
+					})
+				}
+				if rawEnvoyExtAuthzHTTP, ok := extProvider["envoyExtAuthzHttp"]; ok {
+					if envoyExtAuthzHTTP, ok := rawEnvoyExtAuthzHTTP.(map[string]interface{}); ok {
+						extProviderName := extProvider["name"].(string)
+						envoyExtAuthzHTTPConfig := &v2.ExtensionProviderEnvoyExternalAuthorizationHTTPConfig{}
+						if service, ok := envoyExtAuthzHTTP["service"]; ok {
+							envoyExtAuthzHTTPConfig.Service = service.(string)
+						} else {
+							return fmt.Errorf("extension provider envoyExtAuthzHTTP '%s' must specify field 'service'", extProviderName)
```

Comment:
Can we add this requirement to the CRD schema?

Reply:
Done